framework/base/src/types/interaction/tx_data.rs

```diff
@@ -13,6 +123,12 @@ where
     fn is_no_call(&self) -> bool;
 
     fn to_call_data_string(&self) -> ManagedBuffer<Env::Api>;
+
+    fn get_code(&self) -> ManagedBuffer<Env::Api>;
```

Comment:
No. This trait is general, it is not allowed to know about anything deploy related.